src/routes/adminPanel/adminPanelTable/tablePagination.component.tsx

```diff
@@ -0,0 +1,72 @@
+interface TablePaginationProps {
+  gotoPage: (page: number) => void;
+  previousPage: () => void;
+  nextPage: () => void;
+  canPreviousPage: boolean;
+  canNextPage: boolean;
+  pageCount: number;
+  pageIndex: number;
+  pageOptions: number[];
+  pageSize: number;
+  setPageSize: (size: number) => void;
+}
+
+export const TablePagination = ({
+  gotoPage,
+  previousPage,
+  nextPage,
+  canPreviousPage,
+  canNextPage,
+  pageCount,
+  pageIndex,
+  pageOptions,
+  pageSize,
+  setPageSize,
+}: TablePaginationProps) => {
+  return (
+    <div className="pagination">
+      <button onClick={() => gotoPage(0)} disabled={!canPreviousPage}>
+        {'<<'}
+      </button>{' '}
+      <button onClick={() => previousPage()} disabled={!canPreviousPage}>
+        {'<'}
+      </button>{' '}
+      <button onClick={() => nextPage()} disabled={!canNextPage}>
+        {'>'}
+      </button>{' '}
+      <button onClick={() => gotoPage(pageCount - 1)} disabled={!canNextPage}>
+        {'>>'}
+      </button>{' '}
+      <span>
+        Page{' '}
+        <strong>
+          {pageIndex + 1} of {pageOptions.length}
+        </strong>{' '}
+      </span>
+      <span>
+        | Go to page:{' '}
+        <input
+          type="number"
+          defaultValue={pageIndex + 1}
+          onChange={(e) => {
+            const page = e.target.value ? Number(e.target.value) - 1 : 0;
+            gotoPage(page);
```

Comment:
is there any check if an inserted page is greater than pages?

Reply:
If you put there a value greater than allowed, basically nothing will happen - the table will show the data from the last pagination.